internal/driver/checkstatuses.go

```diff
@@ -85,7 +100,7 @@ func (d *Driver) tcpProbe(device sdkModel.Device) bool {
 	return true
 }
 
-func (d *Driver) updateDeviceStatus(deviceName string, status string) error {
+func (d *Driver) updateDeviceStatus(deviceName string, status string) (bool, error) {
```

Comment:
add a function comment to say that it returns whether or not the device status changed, and any errors that ocurred.